lib/date.js

```diff
@@ -1,8 +1,13 @@
 import { truncate } from './helpers'
 
 export default function inspectDate(dateObject, options) {
-  // If we need to - truncate the time portion, but never the date
-  const split = dateObject.toJSON().split('T')
-  const date = split[0]
-  return options.stylize(`${date}T${truncate(split[1], options.truncate - date.length - 1)}`, 'date')
+  const stringRepresentation = dateObject.toJSON()
+  if (stringRepresentation === null) {
+    return 'Invalid Date'
```

Comment:
Do you think checking `NaN` here would be better?

```suggestion
  if (Number.isNaN(dateObject.valueOf())) {
    return 'Invalid Date'
```

Reply:
Good question. In my test both SpiderMonkey and V8 returns `null` for `.toJSON()`, so calling `Number.isNaN` is just a needless extra call. But if there are implementations out there that returns anything else when `.toJSON()` is called on an invalid `Date` object, then yes - `NaN` is better.

However, [21.4.4.37 Date.prototype.toJSON ( key )](https://262.ecma-international.org/#sec-date.prototype.tojson) might be doing those steps (`Number.isNaN(dateObject.valueOf())`) anyway. Step 2 is _"Let tv be ? [ToPrimitive](https://262.ecma-international.org/#sec-toprimitive)(O, number)"_ which will call `.valueOf()` but step 3 confuses me.

Reply:
`dateObject.valueOf()` will always return a `Number` for a `Date` object - `NaN` is also a `Number` type.

1. `dateObject.toJSON()` calls `valueOf()` and checks that, a `Number` is finite - so that is an extra check we get for free*.
    1. `Number.isFinite(NaN)` and `Number.isFinite(9e999)` are both `false` for example.
1. It then calls `toISOString()` which always throws a `RangeError: invalid date`, if somehow the above check was not enough and will return `null` in that case.

_\* since we need to call `.toJSON()` anyway to get the serialization to work_

In short, I don't think `Number.isNaN(dateObject.valueOf())` is better than `dateObject.toJSON() === null`, since we have to do the latter anyway.